### PR TITLE
initial infra project structure for SUMO, S3 + Cloudfront distributions

### PR DESCRIPTION
**Pull request by metadave**
- create a directory structure for SUMO similar to all of our other apps in the infra repo
- provision stage and prod user media buckets for use with Cloudfront
- provision Cloudfront distributions for stage and prod media buckets
- I used the values `stage-cdn.sumo.mozilla.net`, `stage-cdn.sumo.moz.works` for stage, and  `prod-cdn.sumo.mozilla.net`, `prod-cdn.sumo.moz.works` for prod CDN domain names. We can use `.sumo.moz.works` for any new SUMO DNS entries. This is similar to MDN, where all new DNS entries are allocated under `mdn.moz.works`. If these domains are ok, I'll need to request 2 certs in `us-east-1` for the Cloudfront distributions (marked as TODO in the Terraform code).

- Cloudfront will automatically redirect http -> https
- S3 logs are stored in the `sumo-user-media-logs` bucket organized by `stage-logs/` and `prod-logs/` directories
- please verify Cloudfront min/max/default TTLs

See also:
https://github.com/mozilla/kitsune/issues/2934
https://github.com/mozilla/kitsune/issues/2933
https://github.com/mozilla/kitsune/issues/2899

### CODE REVIEW

File: apps/sumo/infra/shared/main.tf [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    logging_bucket_id = "${aws_s3_bucket.logs.id}"
+    logging_prefix = "prod-logs/"
+    region = "${var.region}"
+    source = "./s3"
+}
+
+module "sumo-user-media-stage-cf" {
+    source = "./cloudfront"
+
+    acm_cert_arn = "TODO"
```
Comment (jgmize) [inline apps/sumo/infra/shared/main.tf]: Let me know once this cert has been requested and I'll approve it.

File: apps/sumo/infra/shared/main.tf [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    aliases = ["stage-cdn.sumo.mozilla.net", "stage-cdn.sumo.moz.works"]
+    comment = "Stage CDN for SUMO user media"
+    distribution_name = "SUMOMediaStageCDN"
+    domain_name = "sumo-user-media-stage.s3-website-us-west-2.amazonaws.com"
+}
+
+module "sumo-user-media-prod-cf" {
+    source = "./cloudfront"
+
+    acm_cert_arn = "TODO"
```
Comment (jgmize) [inline apps/sumo/infra/shared/main.tf]: same here

**Comment by glogiotatidis** [comment]
> I used the values stage-cdn.sumo.mozilla.net, stage-cdn.sumo.moz.works for stage, and prod-cdn.sumo.mozilla.net, prod-cdn.sumo.moz.works for prod CDN domain names. We can use .sumo.moz.works for any new SUMO DNS entries. This is similar to MDN, where all new DNS entries are allocated under mdn.moz.works. If these domains are ok, I'll need to request 2 certs in us-east-1 for the Cloudfront distributions (marked as TODO in the Terraform code).

Domains are fine. Please also add `-dev`

**Comment by glogiotatidis** [comment]
> please verify Cloudfront min/max/default TTLs

These look fine for now. I'm sure we'll revisit in the future.

**Comment by metadave** [comment]
dev added, certs all set. 

I'll submit a bug to have CNAMEs setup for:

- `dev-cdn.sumo.mozilla.net` -> `dev-cdn.sumo.moz.works`
- `stage-cdn.sumo.mozilla.net` -> `stage-cdn.sumo.moz.works`
- `prod-cdn.sumo.mozilla.net` -> `prod-cdn.sumo.moz.works`

**Comment by metadave** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=1413279

**Comment by metadave** [comment]
Terraform has NOT been applied to save $$.

**Comment by metadave** [comment]
DNS has been setup per https://bugzilla.mozilla.org/show_bug.cgi?id=1413279

**Pull request merged**